src/riemannian/hamiltonian.jl

```diff
@@ -72,9 +72,10 @@ function step(
         end
         #! Eq (17) of Girolami & Calderhead (2011)
         θ_full = copy(θ_init)
-        term_1 = ∂H∂r(h, θ_init, r_half) # unchanged across the loop
+        term_1 = ∂H∂r(h, θ_init, r_half).gradient # unchanged across the loop
```

Comment:
This computes `neg_energy` but doesn't use it, which has performance implications. Perhaps change to 
```suggestion
        term_1 = ∂H∂r(h, r_half) # unchanged across the loop
```

Reply:
Ohh, I suddenly realize the riemannian manifold HMC is not included in AdvancedHMC yet, this file is not included in the package, see https://github.com/TuringLang/AdvancedHMC.jl/blob/5a562e0472fab1892d59cb4cca3090124903acb5/src/AdvancedHMC.jl#L50

I think we might start with #439, which is ready to get in now.